cluster/kubernetes/manifests.go

```diff
@@ -7,12 +7,36 @@ import (
 	"github.com/weaveworks/flux/resource"
 )
 
+// namespacer assigns namespaces to manifests that need it (or "" if
+// the manifest should not have a namespace.
+type namespacer interface {
+	correctNamespace(kresource.KubeManifest) error
+}
+
+// Manifests is an implementation Manifests particular to
+// Kubernetes. Aside from loading manifests from files, it does some
+// "post-processsing" to make sure the view of the manifests is what
+// would be applied; in particular, it fills in the namespace of
+// manifests that would be given a default namespace when applied.
 type Manifests struct {
-	FallbackNamespace string
+	Namespacer namespacer
 }
 
 func (c *Manifests) LoadManifests(base string, paths []string) (map[string]resource.Resource, error) {
-	return kresource.Load(base, c.FallbackNamespace, paths)
+	manifests, err := kresource.Load(base, paths)
+	if err != nil {
+		return nil, err
+	}
+	result := map[string]resource.Resource{}
+	for _, km := range manifests {
+		if c.Namespacer != nil {
+			if err = c.Namespacer.correctNamespace(km); err != nil {
+				return nil, err // TODO(michael) Consider whether it's better to just omit this resource
```

Comment:
I think it's better to report an error, since simply omitting the resource can lead to an unexpected behavior (like not syncing resources _atomically_ (or in one go)).